code/modules/mob/living/carbon/xenomorph/castes/puppeteer/abilities_puppeteer.dm

```diff
@@ -0,0 +1,466 @@
+		detonate(puppet)
+
+///detonates a puppet causing a spray of acid
+/datum/action/xeno_action/activable/organic_bomb/proc/detonate(mob/living/puppet)
+	SIGNAL_HANDLER
+	UnregisterSignal(puppet, list(COMSIG_XENOMORPH_ATTACK_LIVING, COMSIG_MOB_DEATH))
+	if(QDELETED(puppet))
+		return
+	puppet.visible_message(span_danger("[puppet] ruptures, releasing corrosive acid!"))
+	var/turf/our_turf = get_turf(puppet)
+	playsound(our_turf, 'sound/bullets/acid_impact1.ogg', 50, 1)
+	puppet.gib()
+
+	for(var/turf/acid_tile AS in RANGE_TURFS(1, our_turf))
+		new /obj/effect/temp_visual/acid_splatter(acid_tile) //SFX
+		if(!locate(/obj/effect/xenomorph/spray) in acid_tile.contents)
+			new /obj/effect/xenomorph/spray(acid_tile, 10 SECONDS, 16)
+// ***************************************
+// *********** Articulate
+// ***************************************
+/datum/action/xeno_action/activable/articulate
+	name = "Articulate"
+	action_icon_state = "mimicry"
+	desc = "Takes direct control of a Puppet’s vocal chords. Allows you to speak directly through your puppet to the talls."
+	cooldown_timer = 10 SECONDS
+	target_flags = XABB_MOB_TARGET
+	///Whether we should cancel instead of doing the thing when activated
+	var/talking = FALSE
+	///our current target
+	var/mob/living/carbon/active_target
+
+/datum/action/xeno_action/activable/articulate/use_ability(mob/living/victim)
+	if(talking)
+		cancel(owner)
+		return fail_activate()
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	if(!istype(victim, /mob/living/carbon/xenomorph/puppet) || !(victim in huskaction.puppets))
+		victim.balloon_alert(owner, "not our puppet")
+		return fail_activate()
+	owner.balloon_alert(owner, "channeling voice, move or activate to cancel!")
+	active_target = victim
+	RegisterSignal(owner, COMSIG_MOB_SAY, PROC_REF(relay_speech))
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(cancel))
+	RegisterSignal(victim, COMSIG_QDELETING, PROC_REF(cancel))
+	talking = TRUE
+	add_cooldown()
+
+/datum/action/xeno_action/activable/articulate/proc/relay_speech(mob/living/carbon/source, arguments)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(relay_speech_async), active_target, arguments[SPEECH_MESSAGE])
+
+/datum/action/xeno_action/activable/articulate/proc/relay_speech_async(mob/living/carbon/target, text)
+	target.say(text, language = /datum/language/common, forced = "puppeteer articulate ability")
+
+/datum/action/xeno_action/activable/articulate/proc/cancel(atom/target)
+	SIGNAL_HANDLER
+	if(talking)
+		owner.balloon_alert(owner, "cancelled!")
+	talking = FALSE
+	active_target = null
+	UnregisterSignal(owner, list(COMSIG_MOB_SAY, COMSIG_MOVABLE_MOVED, COMSIG_QDELETING))
+
+// ***************************************
+// *********** Living Construct (Primordial)
+// ***************************************
+/datum/action/xeno_action/activable/living_construct
```

Comment:
This whole ability is super wack. You can possess stuff like full fuel tanks, and its literally the googly eyed wizard thing from ss13. I feel like its a better idea to just focus on the puppets. Having this as well seems very unfocused.

Reply:
we dont have this ability anymore